src/kotlin/example/spring/src/main/kotlin/com/planetnine/sessionless/util/sessionless/models/ISessionless.kt

```diff
@@ -0,0 +1,87 @@
+
+interface ISessionless {
+    /** The way to store and retrieve key pairs */
+    val vault: IVault
+
+    interface WithKeyStore : ISessionless {
+        override val vault: KeyStoreVault
+
+        /** Generates a private/public key pair and stores it using [vault]
+         * @param keyAccessInfo Info required to access the stored key
+         * @return The [KeyPair] that was generated then stored with [vault] */
+        fun generateKeys(keyAccessInfo: KeyAccessInfo): KeyPair
+
+        /** Generates a private/public key pair and stores it using [vault]
+         * @param keyAccessInfo Info required to access the stored key
+         * @return The [KeyPair] that was generated then stored with [vault] */
+        suspend fun generateKeysAsync(keyAccessInfo: KeyAccessInfo): KeyPair
+
+        /** Retrieves keys using [vault]
+         * @param keyAccessInfo Info required to access the stored key
+         * @return Key pair as objects ([KeyPair]) */
+        fun getKeys(keyAccessInfo: KeyAccessInfo): KeyPair
+
+        /** Signs a [message] with the user's stored private key (from [vault])
+         * @param message The message to be signed.
+         * @param keyAccessInfo Info required to access the stored key
+         * @return Signature as a [String] */
+        fun sign(message: String, keyAccessInfo: KeyAccessInfo): MessageSignature
+    }
+
+    interface WithCustomVault : ISessionless {
+        override val vault: ICustomVault
+
+        /** Generates a private/public key pair and stores it using [vault]
+         * @return A simplified ([SimpleKeyPair]) version of the key pair that was generated then stored with [vault] */
+        fun generateKeys(): SimpleKeyPair
+
+        /** Generates a private/public key pair and stores it using [vault]
+         * @return A simplified ([SimpleKeyPair]) version of the key pair that was generated then stored with [vault] */
+        suspend fun generateKeysAsync(): SimpleKeyPair
+
+        /** Retrieves keys using [vault]
+         * @return Key pair as [String]s ([SimpleKeyPair]). */
+        fun getKeys(): SimpleKeyPair
+
+        /** Signs a [message] with the user's stored private key (from [vault]).
+         * @param message The message to be signed.
+         * @return Signature as a [MessageSignature] */
+        fun sign(message: String): MessageSignature
+    }
+
+    /** Signs a [message] using the provided [privateKey].
+     * @param message The message to be signed.
+     * @return Signature as a [MessageSignature]. */
+    fun sign(message: String, privateKey: PrivateKey): MessageSignature
+
+    /** Verifies a given signature with a public key.
+     * @param identifiableMessage message that was signed
+     * @return True if the [IdentifiableMessage.signature] is valid
+     * for the given [IdentifiableMessage.message] and [IdentifiableMessage.publicKey].
+     * @see sign */
+    fun verify(identifiableMessage: IdentifiableMessage): Boolean
```

Comment:
This should be `verifySignature` to stay consistent with the other implementations

Reply:
Sure thing

I also noticed an error here 
`IdentifiableMessage` was renamed to `SignedMessage` but this file wasn't updated for some reason